src/main/java/com/example/gradecalculator/controller/UserController.java

```diff
@@ -127,5 +125,20 @@ public String performLogout(Authentication authentication, HttpServletRequest re
         this.logoutHandler.logout(request, response, authentication);
         return "index";
     }
+
+    @DeleteMapping("/myProfile")
+    public ResponseEntity<Void> deleteUser(Authentication authentication,HttpServletRequest request, HttpServletResponse response) {
+        var userId = userService.getAuthenticatedUserId(authentication);
+
+        boolean profileDeleted = userService.deleteUserAndSubjects(userId);
```

Comment:
try to start boolean variables with 'is', 'has', 'can', 'was' or 'should'
-> e.g. 'wasDeleted'